mvc/src/test/java/nextstep/mvc/controller/tobe/AnnotationHandlerAdapterTest.java

```diff
@@ -0,0 +1,48 @@
+package nextstep.mvc.controller.tobe;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import nextstep.mvc.view.JspView;
+import nextstep.mvc.view.ModelAndView;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class AnnotationHandlerAdapterTest {
+
+    private AnnotationHandlerAdapter handlerAdapter;
+
+    @BeforeEach
+    void setUp() {
+        handlerAdapter = new AnnotationHandlerAdapter();
+    }
+
+    @Test
+    void support() {
+        final boolean actual = handlerAdapter.supports(mock(HandlerExecution.class));
+
+        assertThat(actual).isTrue();
+    }
+
+    @Test
+    void support_notHandler() {
+        final boolean actual = handlerAdapter.supports(mock(String.class));
+
+        assertThat(actual).isFalse();
+    }
+
+    @Test
+    void handle() throws Exception {
+        final var request = mock(HttpServletRequest.class);
+        final var response = mock(HttpServletResponse.class);
+        final var handler = mock(HandlerExecution.class);
+
+        when(handlerAdapter.handle(request, response, handler)).thenReturn(new ModelAndView(mock(JspView.class)));
+        ModelAndView modelAndView = handlerAdapter.handle(request, response, handler);
+
+        assertThat(modelAndView.getView()).isInstanceOf(JspView.class);
```

Comment:
요 테스트는
mock을 사용해서 handle을 호출했을 때 `new ModelAndView(mock(JspView.class))`를 반환하게 하고
그것이 잘 반환되었는지 확인하는 테스트 같은데요!

```suggestion
    @Test
    void handle() throws Exception {
        AnnotationHandlerAdapter annotationHandlerAdapter = new AnnotationHandlerAdapter();
        AnnotationHandlerMapping annotationHandlerMapping = new AnnotationHandlerMapping("samples");
        annotationHandlerMapping.initialize();
        final var request = mock(HttpServletRequest.class);
        final var response = mock(HttpServletResponse.class);
        when(request.getAttribute("id")).thenReturn("gugu");
        when(request.getRequestURI()).thenReturn("/get-test");
        when(request.getMethod()).thenReturn("GET");

        ModelAndView result = annotationHandlerAdapter.handle(request, response, annotationHandlerMapping.getHandler(request));

        assertThat(result.getObject("id")).isEqualTo("gugu");
    }
```

요런 방식은 어떨까요??

Reply:
오.. 저도 이 테스트가 가장 마음에 걸리던 부분이었는데 좋은 의견 감사합니다!! 추가로 기존 테스트는 verify를 이용해 의도한대로 호출되고 있는지 확인하도록 수정해 보았습니다 :)